torch/_refs/__init__.py

```diff
@@ -2923,6 +2923,13 @@ def narrow(
     torch._check(length >= 0, lambda: "narrow(): length must be non-negative.")
     dim = utils.canonicalize_dim(a.ndim, dim)
     dim_length = a.size(dim)
+    torch._check_with(
+        IndexError,
+        -dim_length <= start and start <= dim_length,  # type: ignore[arg-type]
+        lambda: f"start out of range (expected to be in range of [{-dim_length}, {dim_length}], but got {start})",
+    )
+    if start < 0:
```

Comment:
isn't this case handled by `canonicalize_dim`?

Reply:
Nope: canonicalize_dim is called on dim, not start ;)

Reply:
I was referring to the `canonicalize_dim` below..

https://github.com/pytorch/pytorch/blob/1804469c3447f605485deeb4f19856e992ee0b72/torch/_refs/__init__.py#L2936-L2940

Reply:
oh oops I need to delete the case below haha

Reply:
Could you elaborate why this change is needed? Is it formulating sym expressions differently?

Reply:
This particular change is needed because sometimes we know that an unbacked symint is <= some other side, but we don't know if it is exactly == to the size. If we write the condition so that we don't need to know, that saves us from erroring